### ISSUE TITLE
Mastodon needs a way to set all past posts to private

### ISSUE THREAD

**Issue by KevGP** [mastodon/mastodon#20589]
### Pitch

Mastodon, in my opinion, needs a way to set all past posts to private or public or unlisted. It currently has a setting to change default privacy to either one of these 3 settings under preferences > other, but that only affects posts going forward. Not past posts.

There should be a way to at least block all non-followers from viewing posts upon request. The "follow request" feature still lets anyone view your posts if you've set them to public, whereas other sites allow you to block strangers from looking altogether.

### Motivation

It's a privacy concern. Newbies might not know how follow requests actually just block people from having their stuff in their home timelines. Stalking one's profile technically can happen if someone gets an account on a new instance the person being stalked doesn't know about and hasn't blocked.

**Comment by trwnh** [comment]
unfortunately this is a bit like asking to unsend an email possibly years after it's already been sent. assuming every remote server understood the Update activity, you could send an Update for every single status to every single known server -- but this would be a really expensive operation.

the only real way around this, if you want to have *future* control, is to stop delivering over federation and instead browse profiles natively on their origin website. you'd also need some cross-domain authentication so that the remote site knows who you are and what you're allowed to see. but then this wouldn't be "public" anymore, so you would probably also need an extension to activitypub to signal a sort of "authenticated only" scope.

**Comment by nemobis** [comment]
This feature already exists and it's called deleting your old posts: visit /statuses_cleanup on your Mastodon instance's web interface to set it up.

Note that deleting old posts is a computationally heavy process which can fail; https://github.com/mastodon/mastodon/pull/22273 attempts to improve the chances of success.

**Comment by trwnh** [comment]
@nemobis deleting old statuses isn't the aim of this issue, this issue is explicitly about preserving the posts but limiting their audience after-the-fact (which is impossible if you federate them, and only possible if fediverse software switches to viewing posts on the origin instance, so that the origin instance can enforce access control)

**Comment by nemobis** [comment]
> which is impossible if you federate them

Yes, so if you've federated them you must first delete the federated copies.

**Comment by trwnh** [comment]
deleting the federated copies and making them private would mean no one can see them except for local followers. this is not the solution you think it is
